content/cpi-api-v2.md

```diff
@@ -0,0 +1,115 @@
+# Cloud Provider Interface
+
+For an overview of the sequence of CPI calls, the following resources are helpful:
+
+- [BOSH components](bosh-components.md) and its example component interaction diagram
+- [CLI v2 architecture doc](https://github.com/cloudfoundry/bosh-cli/blob/master/docs/architecture.md#deploy-command-flow) and [`bosh create-env` flow](https://github.com/cloudfoundry/bosh-init/blob/master/docs/init-cli-flow.png) where calls to the CPI are marked as `cloud`.
+
+Examples of API request and response:
+
+- [Building a CPI: RPC - Request](https://bosh.io/docs/build-cpi.html#request)
+- [Building a CPI: RPC - Response](https://bosh.io/docs/build-cpi.html#response)
+
+
+Library:
+
+- Ruby: `bosh-cpi-ruby` gem [v2.5.0](https://github.com/cloudfoundry/bosh-cpi-ruby/releases/tag/v2.5.0)
+- GoLang: `bosh-cpi-go` [library](https://github.com/cppforlife/bosh-cpi-go)
+
+---
+
+## Glossary {: #glossary }
+
+- **cloud ID** is an ID (string) that the Director uses to reference any created infrastructure resource; typically CPI methods return cloud IDs and later receive them. For example AWS CPI's `create_vm` method would return `i-f789df` and `attach_disk` would take it.
+
+- **cloud_properties** is a hash that can be specified for several objects (resource pool, disk pool, stemcell, network) to provide infrastructure specific settings to the CPI for that object. Only CPIs know the meaning of its contents. For example resource pool's `cloud_properties` for AWS can specify `instance_type`:
+
+```yaml
+resource_pools:
+- name: large_machines
+  cloud_properties: {instance_type: r3.8xlarge}
+```
+
+## Methods
+
+- All V1 contracts must still be supported. See [CPI API V1](cpi-api-v1.md).
+- To differentiate V2 calls the caller needs to pass in `"api_version": 2` in the context
+
+### Reference Table (Based on each component version)
+* Reg./reg. : Registry
```

Comment:
Can we drop the legend and instead write 'registry' everywhere?

Reply:
Will update.